arch/arm/core/swap.S

```diff
@@ -47,12 +47,18 @@ GDATA(_kernel)
 
 SECTION_FUNC(TEXT, __pendsv)
 
+#if defined (CONFIG_KERNEL_EVENT_LOGGER_CONTEXT_SWITCH) || \
+		defined(CONFIG_STACK_SENTINEL)
+    /* Register the context switch */
+    push {lr}
+
 #ifdef CONFIG_KERNEL_EVENT_LOGGER_CONTEXT_SWITCH
-	/* Register the context switch */
-	push {lr}
-	bl _sys_k_event_logger_context_switch
-	pop {r0}
-	mov lr, r0
+    bl _sys_k_event_logger_context_switch
+#endif
+#ifdef CONFIG_STACK_SENTINEL
+    bl _check_stack_sentinel
+#endif
+    pop {lr}
```

Comment:
Perhaps ARMv6-M (Cortex-M0) limitation? If byte/clock-saving is important, perhaps a conditional would help.